build-hook/src/main.rs

```diff
@@ -22,9 +23,17 @@ fn parse_drv_hash<'a>(drv_path: &'a str) -> &'a str {
     re.captures(drv_path).unwrap().get(1).unwrap().as_str()
 }
 
+fn fingerprint(out_path: &str, nar_hash: &str, size: u64) -> String {
+    let references = query_references(out_path).expect("Query references").join(",");
+    let fingerprint = format!("1;{out_path};{nar_hash};{size};{references}").to_string();
```

Comment:
have not verified this yet, but looks plausible:

```
1;/nix/store/6j0m4armmz899d9sxjqrl2wb5la777sc-unit-nix-daemon.service;sha256:1y8nx48i4i7kpi65myw80zfcgbr8rf1w25lzkwg95j5rplfc1zlx;1864;/nix/store/4vy9ahhyq5q6cgqi5054s7fvgnrxrdwi-glibc-locales-2.38-44,/nix/store/bicmg5gd50q6igk0y5mga1v0p1lk8f26-coreutils-9.4,/nix/store/dwfm7k4s037k0v9zljabzdmndmsfgy84-tzdata-2024a,/nix/store/m0nicpa4d724rljadlmw1m3qv0ja4syy-systemd-255.2,/nix/store/mn911d51n5lklwr3zy4mdhxa77wzancb-gnugrep-3.11,/nix/store/nx7wqwfsxx6f2n7vf3v28y4adyvnin85-X-Restart-Triggers-nix-daemon,/nix/store/nzkd2mf4792r7aia5fw3f7libpk2rxyj-nix-2.18.1,/nix/store/p6fd7piqrin2h0mqxzmvyxyr6pyivndj-findutils-4.9.0,/nix/store/sdby9a3m1dciynkjbi7xj7pk6dsilmw4-openssh-9.6p1,/nix/store/sqi0c52ba09g9rymng9lrk8qx7iw0z9z-util-linux-2.39.3-bin,/nix/store/vd92lhcxs39hbdnzj8ycak5wvj466s3l-gnused-4.9
```

To be confirmed: should the `/nix/store` and `sha256:` prefixes be there?